src/app/pages/system/alert-config/alert-config.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { FormBuilder } from '@ngneat/reactive-forms';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { TranslateService } from '@ngx-translate/core';
+import { of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { AlertLevel } from 'app/enums/alert-level.enum';
+import { AlertPolicy } from 'app/enums/alert-policy.enum';
+import helptext from 'app/helptext/system/alert-settings';
+import { AlertCategory, AlertClassesUpdate, AlertClassSettings } from 'app/interfaces/alert.interface';
+import { WebsocketError } from 'app/interfaces/websocket-error.interface';
+import { EntityUtils } from 'app/modules/entity/utils';
+import { SnackbarService } from 'app/modules/snackbar/services/snackbar.service';
+import { DialogService, WebSocketService } from 'app/services/';
+
+@UntilDestroy()
+@Component({
+  selector: 'ix-alert-config',
+  templateUrl: './alert-config.component.html',
+  styleUrls: ['./alert-config.component.scss'],
+})
+export class AlertConfigComponent implements OnInit {
+  categories: AlertCategory[] = [];
+  selectedCategoryId = '';
+  form = this.formBuilder.group({});
+  isFormLoading = false;
+  readonly helptext = helptext;
+
+  readonly levelOptions$ = of([
+    { label: this.translate.instant('INFO'), value: AlertLevel.Info },
+    { label: this.translate.instant('NOTICE'), value: AlertLevel.Notice },
+    { label: this.translate.instant('WARNING'), value: AlertLevel.Warning },
+    { label: this.translate.instant('ERROR'), value: AlertLevel.Error },
+    { label: this.translate.instant('CRITICAL'), value: AlertLevel.Critical },
+    { label: this.translate.instant('ALERT'), value: AlertLevel.Alert },
+    { label: this.translate.instant('EMERGENCY'), value: AlertLevel.Emergency },
+  ]);
+
+  readonly policyOptions$ = this.ws.call('alert.list_policies').pipe(
+    map((policyList) => {
+      return policyList.map((policy) => ({ label: policy, value: policy }));
+    }),
+  );
+
+  get selectedCategory(): AlertCategory {
+    return this.categories.find((category) => this.selectedCategoryId === category.id);
```

Comment:
UI fetches this value hundreds of times per second sometimes. Which means, we're running this check on this array thousands of times as you can see in this attachment. I add a console log `checking` inside the find and you can see how many times we're checking array items for no reason by the log counter. We should keep the selected category value in one variable and just return that variable to UI. Running a find method inside this getter is very inefficient. On checkbox value change, just update the variable value to the full category instead of just the `selectedCategoryId`.

https://user-images.githubusercontent.com/24204846/188987353-2b884d62-af3a-4479-bfe8-d06f5f75afea.mov